staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -336,27 +339,22 @@ func addEmbeddedProperties(s *spec.Schema) {
 	if s.Items != nil {
 		addEmbeddedProperties(s.Items.Schema)
 	}
+	if s.AdditionalProperties != nil {
+		addEmbeddedProperties(s.AdditionalProperties.Schema)
+	}
 	if isTrue, ok := s.VendorExtensible.Extensions.GetBool("x-kubernetes-embedded-resource"); ok && isTrue {
+		s.SetProperty("metadata", *spec.RefSchema(objectMetaSchemaRef).
+			WithDescription(swaggerPartialObjectMetadataDescriptions["metadata"]))
 		addTypeMetaProperties(s)
-
+		// add client-side requirement for kind and apiVersion. This doesn't hurt
+		// TODO(roycaihw, 1.16): add the same requirement to builtin spec, to be consistent
 		req := sets.NewString(s.Required...)
-
 		if !req.Has("kind") {
 			s.Required = append(s.Required, "kind")
 		}
-		kindProp := s.Properties["kind"]
```

Comment:
FYI, I customized the descriptions because they don't apply exactly when used as embedded. For example, kind is not immutable.

Reply:
got it. I will add it back